src/commonmark.c

```diff
@@ -162,6 +162,8 @@ static cmark_node *get_containing_block(cmark_node *node) {
   return NULL;
 }
 
+static unsigned int footnote_ix;
```

Comment:
Proooobably not fine. 😄

It seems like this single variable stops the whole library from being threadsafe. Can we give this some more thought? Surely the footnote index can fit in one of the many state structs we're already passing around.

Reply:
TOTALLY FINE.

(Shit, yeah.)